apps/kafka-mapper-microfrontend/src/components/ConfigEditor/AddExpressionDialog.tsx

```diff
@@ -0,0 +1,139 @@
+import { useState } from "react";
+import {
+  Button,
+  Stack,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  TextField,
+  Link,
+} from "@mui/material";
+import fhirpath from "fhirpath";
+import { useFhirMapperConfig } from "../FhirMapperConfigProvider";
+
+export function AddExpressionDialog() {
+  const [open, setOpen] = useState(false);
+  const [previewValue, setPreviewValue] = useState("");
+  const [expression, setExpression] = useState("");
+  const [columnName, setColumnName] = useState("");
+  const { addExpression, activeFhirResource } = useFhirMapperConfig();
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+    setPreviewValue("");
+  };
+
+  const handleAddElement = (event) => {
+    event.preventDefault();
+    addExpression({
+      resourceType: activeFhirResource.resourceType,
+      columnName: columnName,
+      fhirPath: expression,
+    });
+    handleClose();
+  };
+
+  function renderPreview() {
+    // evaluate FHIRPath expression on FHIR resource and return result
+    // TODO : add handle errors
```

Comment:
Pending

Reply:
Yes, same thing here, we should improve FHIRPath error messages to be more user-friendly. This will help users identify the specific part of their expression that is causing the error.